### ISSUE TITLE
tslint give no warning when use await in a for...loop.

### ISSUE THREAD

**Issue by mrdulin** [palantir/tslint#3767]
### Bug Report

- __TSLint version__:   "tslint": "^5.5.0"
- __TypeScript version__:  "typescript": "^2.7.0-dev.20171108"
- __Running TSLint via__:  VSCode

#### TypeScript code being linted

```ts
// code snippet
const fetch = () =>
  new Promise((resolve, reject) => {
    setTimeout(() => {
      resolve(Math.random());
    }, 1000);
  });

const getRandomNum = async () => {
  for (let i = 0; i < 3; i++) {
    const res = await fetch();
    console.log(res);
  }
};

getRandomNum();
```

with `tslint.json` configuration:

```json
{
  "extends": "tslint:recommended",
  "rules": {
    "object-literal-sort-keys": false,
    "object-literal-key-quotes": [true, "as-needed"],
    "quotemark": [false],
    "trailing-comma": [
      true,
      {
        "multiline": "never",
        "singleline": "never"
      }
    ],
    "ordered-imports": [false],
    "no-console": [false],
    "max-line-length": [true, 150],
    "no-consecutive-blank-lines": false,
    "no-empty": false,
    "no-empty-interface": false,
    "no-trailing-whitespace": false,
    "only-arrow-functions": [false],
    "space-before-function-paren": false,
    "curly": false,
    "no-namespace": false,
    "no-reference": false,
    "no-var-requires": false,
    "comment-format": [false],
    "arrow-parens": false
  }
}
```

#### Actual behavior

tslint give no warning

#### Expected behavior

should like `eslint`, give warning like this:

```
[eslint] Unexpected `await` inside a loop. (no-await-in-loop)
```

**Comment by suchanlee** [comment]
why would you expect tslint to give you warnings in this case?

**Comment by suchanlee** [comment]
I don't think it's the responsibility of tslint to keep track of where and how you use your awaits. The above is valid code and depending on what the context is it may or may not be correct. tslint helps with readability, maintainability, and functionality errors, none of which I believe include this particular issue.

Closing but happy to revisit if you feel strongly otherwise.

**Comment by dgreene1** [comment]
For future readers, `for of` supports blocking for each promise to resolve.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

So use that instead of `for` or `foreach`

**Comment by jordanbtucker** [comment]
@dgreene1 Why do you recommend using `for...of` over `for`? They both block with `await` in same way.

**Comment by dgreene1** [comment]
@jordanbtucker, you are correct. I’ve updated my comment above.

I think I had been taking the promise.all approach so often that I forgot that the regular `for` does the trick. So when I saw that tslint wasn’t picking up the “issue” I assumed that it had to do with the choice of loop. After looking up the original issue I realized that the lint rule the OP was asking for had more to do with a reminder that calls can often be parallelized. 

Anyway, thank you for the correction @jordanbtucker.
